packages/components/src/internal/actions.ts

```diff
@@ -1813,126 +1844,59 @@ function applySelection(
     };
 }
 
-export function initLookup(column: QueryColumn, maxRows: number, values?: List<string>, keys?: List<any>) {
-    if (shouldInitLookup(column, values)) {
-        const store = new LookupStore({
-            key: LookupStore.key(column),
-            isLoaded: false,
-            isLoading: true,
-        });
-        updateLookupStore(store, {}, false);
+export const findLookupValues = async (
+    column: QueryColumn,
+    lookupKeyValues: any[],
+    lookupValues?: any[]
+): Promise<{ column: QueryColumn; descriptors: ValueDescriptor[] }> => {
+    const lookup = column.lookup;
+    const selectRowsOptions: any = {
+        schemaName: lookup.schemaName,
+        queryName: lookup.queryName,
+        columns: [lookup.displayColumn, lookup.keyColumn].join(','),
+        containerPath: lookup.containerPath,
+        maxRows: (lookupKeyValues?.length || 0) + (lookupValues?.length || 0),
```

Comment:
any reason not to just make this maxRows: -1 (or whatever values means give me all the matching rows)?

Reply:
I guess not.  If you don't pass in either values or keyValues, you'll get ALL values, but that's really not a reasonable case here.